tools/nni_cmd/config_schema.py

```diff
@@ -20,267 +20,307 @@
-'experimentName': str,
-Optional('description'): str,
-'trialConcurrency': And(int, lambda n: 1 <=n <= 999999),
-Optional('maxExecDuration'): Regex(r'^[1-9][0-9]*[s|m|h|d]$'),
-Optional('maxTrialNum'): And(int, lambda x: 1 <= x <= 99999),
-'trainingServicePlatform': And(str, lambda x: x in ['remote', 'local', 'pai', 'kubeflow', 'frameworkcontroller']),
-Optional('searchSpacePath'): os.path.exists,
-Optional('multiPhase'): bool,
-Optional('multiThread'): bool,
-Optional('nniManagerIp'): str,
-Optional('logDir'): os.path.isdir,
-Optional('debug'): bool,
-Optional('logLevel'): Or('trace', 'debug', 'info', 'warning', 'error', 'fatal'),
-Optional('logCollection'): Or('http', 'none'),
-'useAnnotation': bool,
-Optional('advisor'): Or({
-    'builtinAdvisorName': Or('Hyperband'),
-    'classArgs': {
-        'optimize_mode': Or('maximize', 'minimize'),
-        Optional('R'): int,
-        Optional('eta'): int
-    },
-    Optional('gpuNum'): And(int, lambda x: 0 <= x <= 99999),
-},{
-    'codeDir': os.path.exists,
-    'classFileName': str,
-    'className': str,
-    Optional('classArgs'): dict,
-    Optional('gpuNum'): And(int, lambda x: 0 <= x <= 99999),
-},{
-    'builtinAdvisorName': Or('BOHB'),
-    'classArgs': {
-        'optimize_mode': Or('maximize', 'minimize'),
-        Optional('min_budget'): And(int, lambda x: 0 <= x <= 9999),
-        Optional('max_budget'): And(int, lambda x: 0 <= x <= 9999),
-        Optional('eta'): And(int, lambda x: 0 <= x <= 9999),
-        Optional('min_points_in_model'): And(int, lambda x: 0 <= x <= 9999),
-        Optional('top_n_percent'): And(int, lambda x: 1 <= x <= 99),
-        Optional('num_samples'): And(int, lambda x: 1 <= x <= 9999),
-        Optional('random_fraction'): And(float, lambda x: 0.0 <= x <= 9999.0),
-        Optional('bandwidth_factor'): And(float, lambda x: 0.0 <= x <= 9999.0),
-        Optional('min_bandwidth'): And(float, lambda x: 0.0 <= x <= 9999.0)
-    },
-    Optional('gpuNum'): And(int, lambda x: 0 <= x <= 99999),
-},{
-    'codeDir': os.path.exists,
-    'classFileName': str,
-    'className': str,
-    Optional('classArgs'): dict,
-    Optional('gpuNum'): And(int, lambda x: 0 <= x <= 99999),
-}),
-Optional('tuner'): Or({
-    'builtinTunerName': Or('TPE', 'Anneal', 'SMAC', 'Evolution'),
-    Optional('classArgs'): {
-        'optimize_mode': Or('maximize', 'minimize')
-    },
-    Optional('includeIntermediateResults'): bool,
-    Optional('gpuNum'): And(int, lambda x: 0 <= x <= 99999),
-},{
-    'builtinTunerName': Or('BatchTuner', 'GridSearch', 'Random'),
-    Optional('gpuNum'): And(int, lambda x: 0 <= x <= 99999),
-},{
+'authorName': setType('authorName', str),
+'experimentName': setType('experimentName', str),
+Optional('description'): setType('description', str),
+'trialConcurrency': setNumberRange('trialConcurrency', int, 1, 99999),
+Optional('maxExecDuration'): And(Regex(r'^[1-9][0-9]*[s|m|h|d]$',error='ERROR: maxExecDuration format is [digit]{s,m,h,d}')),
```

Comment:
Please optimize the indent of this entire file.

Reply:
updated, use indent before line.